client/consensus/babe/src/authorship.rs

```diff
@@ -241,16 +240,26 @@ fn claim_primary_slot(
 		// be empty.  Therefore, this division in `calculate_threshold` is safe.
 		let threshold = super::authorship::calculate_primary_threshold(c, authorities, *authority_index);
 
-		let pre_digest = get_keypair(pair)
-			.vrf_sign_after_check(transcript, |inout| super::authorship::check_primary_threshold(inout, threshold))
-			.map(|s| {
-				PreDigest::Primary(PrimaryPreDigest {
-					slot_number,
-					vrf_output: VRFOutput(s.0.to_output()),
-					vrf_proof: VRFProof(s.1),
-					authority_index: *authority_index as u32,
-				})
-			});
+		let (output, proof) = keystore.read().vrf_sign(
+			pair.as_ref(), &BABE_ENGINE_ID, randomness, slot_number, *epoch_index,
+		);
+		let proof = schnorrkel::vrf::VRFProof::from_bytes(&proof).ok()?;
+		let output = schnorrkel::vrf::VRFOutput::from_bytes(&output).ok()?;
+		let inout = output.attach_input_hash(
+			&pair.as_ref().as_ref().public,
+			transcript
```

Comment:
Note to self: transcript created above should be either:
1. passed to the keystore, OR
2. the keystore creates it and returns it